src/stream.jl

```diff
@@ -190,43 +194,55 @@ function Base.close(stream::TranscodingStream)
 end
 
 function Base.eof(stream::TranscodingStream)
-    mode = stream.state.mode
-    if mode == :idle
-        changemode!(stream, :read)
-        return eof(stream)
-    elseif mode == :read
-        return buffersize(stream.state.buffer1) == 0 && fillbuffer(stream) == 0
-    elseif mode == :write
-        return eof(stream.stream)
-    elseif mode == :close
-        return true
-    elseif mode == :stop
-        return buffersize(stream.state.buffer1) == 0
-    elseif mode == :panic
-        throw_panic_error()
+    eof = buffersize(stream.buffer1) == 0
+    state = stream.state
+    if (state.mode == :read || state.mode == :close) && !eof
```

Comment:
Should this be `:stop` instead of `:close`? If `state.mode` was `:close`, `eof`  returned `true` before.

Reply:
```suggestion
    if (state.mode == :read || state.mode == :stop) && !eof
```